frontend/src/static/js/v_authorship.js

```diff
@@ -224,24 +270,30 @@ window.vAuthorship = {
     },
 
     updateFilterSearch(evt) {
-      if (this.filterType === 'checkboxes') {
-        this.indicateSearchBar();
-      }
       this.filterSearch = (evt.target.value.length !== 0) ? evt.target.value : '*';
     },
 
+    updateGlobHash() {
+      window.addHash('authorshipFilesGlob', this.filterSearch);
+      window.removeHash('authorshipFileTypes');
+      window.encodeHash();
+    },
+
+    updateFileTypeHash() {
+      const fileTypeHash = this.selectedFileTypes.length > 0
+          ? this.selectedFileTypes.reduce((a, b) => `${a}~${b}`)
+          : '';
+      window.addHash('authorshipFileTypes', fileTypeHash);
+      window.removeHash('authorshipFilesGlob');
+      window.encodeHash();
+    },
+
     indicateSearchBar() {
-      this.selectedFileTypes = this.fileTypes.slice();
       this.filterType = 'search';
     },
 
     indicateCheckBoxes() {
-      if (this.filterType === 'search') {
-        const searchBar = document.getElementById('search');
-        searchBar.value = '';
-        this.filterSearch = '*';
-        this.filterType = 'checkboxes';
-      }
+      this.filterType = 'checkboxes';
```

Comment:
I removed this as the filterType watcher already executes this chunk with `updateXxHash` as well, to avoid duplication, likewise for the line in `indicateSearchbar`